addons/web/static/src/core/browser/router_service.js

```diff
@@ -103,61 +111,62 @@ export function parseSearchQuery(search) {
  */
 export function routeToUrl(route) {
     const search = objectToUrlEncodedString(route.search);
-    const hash = objectToUrlEncodedString(route.hash);
-    return route.pathname + (search ? "?" + search : "") + (hash ? "#" + hash : "");
+    return route.pathname + (search ? "?" + search : "");
 }
 
 function getRoute(urlObj) {
     const { pathname, search, hash } = urlObj;
     const searchQuery = parseSearchQuery(search);
-    const hashQuery = parseHash(hash);
-    return { pathname, search: searchQuery, hash: hashQuery };
+    const hashQuery = parseHash(hash); // For retro-compatibility purpose
+    return { pathname, search: { ...searchQuery, ...hashQuery } };
```

Comment:
return { ...searchQuery }
Find another way to do the retro-compatibility, to allow it to re-write the url